### PR TITLE
Topic/sof dev cleanup1

### PR DESCRIPTION
**Pull request by lgirdwood**
Cleanup of code for upstreaming, some function reorg to group functions by feature and commets added to some code blocks.
Separated into multiple patches for easier squashing.

### CODE REVIEW

File: sound/soc/sof/ipc.c [hunk 1]
```diff
@@ -6,6 +6,9 @@
  * Copyright(c) 2017 Intel Corporation. All rights reserved.
```
Comment (plbossart) [inline sound/soc/sof/ipc.c:6]: checkpatch issues:
WARNING: 'mesage' may be misspelled - perhaps 'message'?
#92: FILE: sound/soc/sof/ipc.c:158:
+	/* add mesage to transmit list */

WARNING: 'mesage' may be misspelled - perhaps 'message'?
#113: FILE: sound/soc/sof/ipc.c:182:
+	/* send message if HW read and mesage in TX list */

CHECK: Please don't use multiple blank lines
#267: FILE: sound/soc/sof/ipc.c:366:
+
+

WARNING: 'postion' may be misspelled - perhaps 'position'?
#453: FILE: sound/soc/sof/ipc.c:478:
+/* get stream postion IPC - use faster MMIO method if available on platform */

**Comment by lgirdwood** [comment]
Checkpatch done, I'll be doing sparse as a separate PR

File: sound/soc/sof/core.c [hunk 2]
```diff
@@ -306,7 +355,6 @@ static int sof_remove(struct platform_device *pdev)
 {
 	struct snd_sof_dev *sdev = dev_get_drvdata(&pdev->dev);
 
-	snd_soc_unregister_platform(&pdev->dev);
```
Comment (plbossart) [inline sound/soc/sof/core.c]: is this going to work for topic/sof-v4.14?

File: sound/soc/sof/ipc.c [hunk 3]
```diff
@@ -31,11 +34,23 @@
 #include "sof-priv.h"
 #include "ops.h"
 
-/* IPC message timeout (msecs) */
-#define IPC_TIMEOUT_MSECS	300
+/* stop PCM on XRUN - used for pipeline debug */
+#define SOF_DEBUG_XRUN_STOP	0
```
Comment (plbossart) [inline sound/soc/sof/ipc.c]: Should we make this a Kconfig option?

File: sound/soc/sof/core.c (hunk 2 above)
Comment (lgirdwood) [reply sound/soc/sof/core.c]: gah, part of the post 4.18+ component API updates. Will revert for 4.14.

File: sound/soc/sof/ipc.c (hunk 3 above)
Comment (lgirdwood) [reply sound/soc/sof/ipc.c]: Done, and also created a debug option submenu  where stuff like this can go

**Pull request merged**